### PR TITLE
Checkstyle

### PR DESCRIPTION
**Pull request by cichan02**
## PR-Codex overview
This PR updates the Spring Boot application with new features and improvements, including cache configuration, exception handling, and documentation updates.

### Detailed summary
- Added Redis cache configuration with a time-to-live value of 2 minutes.
- Added new exception classes for resource creation and retrieval, with constructors that take a detail message.
- Updated package-info.java files with Javadoc comments.
- Added a new marker interface `OnCreate` for DTO validation.
- Added new MapStruct mappers for converting between DTOs and domain entities.
- Updated the main class with Javadoc comments.
- Added a new method to `MongoCountryRepository` for checking if a country with the specified name and ID exists in the repository.
- Updated `CountryDTO` with a new maximum allowed length for the country's name.
- Updated `CityMapper` with Javadoc comments and added a new method for converting City entities to CityDTOs.

> The following files were skipped due to too many changes: `src/main/java/by/piskunou/solvdlaba/web/mapper/CityMapper.java`, `src/main/java/by/piskunou/solvdlaba/persistence/MongoCity.java`, `src/main/java/by/piskunou/solvdlaba/persistence/MongoAirport.java`, `src/main/java/by/piskunou/solvdlaba/persistence/impl/CountryRepositoryImpl.java`, `src/main/java/by/piskunou/solvdlaba/web/dto/CityDTO.java`, `src/main/java/by/piskunou/solvdlaba/web/mapper/AirportMapper.java`, `src/main/java/by/piskunou/solvdlaba/persistence/impl/CityRepositoryImpl.java`, `src/main/java/by/piskunou/solvdlaba/service/CountryService.java`, `src/main/java/by/piskunou/solvdlaba/persistence/impl/AirportRepositoryImpl.java`, `src/main/java/by/piskunou/solvdlaba/persistence/CountryRepository.java`, `pom.xml`, `src/main/java/by/piskunou/solvdlaba/service/config/RedisConfig.java`, `src/main/java/by/piskunou/solvdlaba/persistence/mongo/MongoAirportRepository.java`, `src/main/java/by/piskunou/solvdlaba/persistence/mongo/MongoCityRepository.java`, `src/main/java/by/piskunou/solvdlaba/service/CityService.java`, `src/main/java/by/piskunou/solvdlaba/persistence/CityRepository.java`, `src/main/java/by/piskunou/solvdlaba/persistence/AirportRepository.java`, `src/main/java/by/piskunou/solvdlaba/service/AirportService.java`, `src/main/java/by/piskunou/solvdlaba/web/dto/AirportDTO.java`, `src/main/java/by/piskunou/solvdlaba/web/controller/AdviceController.java`, `src/main/java/by/piskunou/solvdlaba/service/impl/CountryServiceImpl.java`, `src/main/java/by/piskunou/solvdlaba/web/controller/CountryController.java`, `src/main/java/by/piskunou/solvdlaba/service/impl/CityServiceImpl.java`, `src/main/java/by/piskunou/solvdlaba/service/impl/AirportServiceImpl.java`, `src/main/java/by/piskunou/solvdlaba/web/controller/CityController.java`, `src/main/java/by/piskunou/solvdlaba/web/controller/AirportController.java`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: src/main/java/by/piskunou/solvdlaba/web/dto/AirportDTO.java [hunk 1]
```diff
@@ -10,30 +10,100 @@
...
 @NoArgsConstructor
 @Schema(description = "Airline info")
 public class AirportDTO {
 
-    @Schema(description = "The airport's unique identification id")
-    @Null(groups = {onCreate.class}, message = "Id should be null")
+    /**
+     * Maximum allowed length of the airport's name.
+     */
+    private static final int NAME_MAX_SIZE = 200;
```
Comment (cichan02) [inline src/main/java/by/piskunou/solvdlaba/web/dto/AirportDTO.java:32]: It was done to avoid 'MagicNumber' from Checkstyle. Is it norm?

**Comment by cichan02** [comment]
Write full Javadocs for this service.

Suppress only auto-generated files.

File: src/main/java/by/piskunou/solvdlaba/domain/Airport.java [hunk 2]
```diff
@@ -20,10 +20,29 @@
...
+     */
     private String name;
+
+    /**
+     * Airport's IATA (International Air Transport Association) code.
+     * For example, for John F. Kennedy International Airport,
+     * this will be JFK.
+     */
     private String iata;
+
```
Comment (h1alexbel) [inline src/main/java/by/piskunou/solvdlaba/domain/Airport.java:40]: nice

File: src/main/java/by/piskunou/solvdlaba/domain/Airport.java [hunk 3]
```diff
@@ -20,10 +20,29 @@
...
+     * Unique identification number in program.
+     */
     @Id
     private String  id;
+
+    /**
+     * Full airport's name. For example, John F. Kennedy International Airport.
+     */
     private String name;
+
```
Comment (h1alexbel) [inline src/main/java/by/piskunou/solvdlaba/domain/Airport.java:33]: nice

File: src/main/java/by/piskunou/solvdlaba/service/impl/CountryServiceImpl.java [hunk 4]
```diff
@@ -19,81 +20,188 @@
...
+     */
     private final ReactiveRedisTemplate<String, Country> template;
 
+    /**
+     * The time-to-live value for Redis cache in minutes,
+     * read from the Spring configuration properties file.
+     */
+    @Value("${spring.cache.redis.time-to-live-in-minutes}")
+    private long ttlInMinutes;
+
```
Comment (h1alexbel) [inline src/main/java/by/piskunou/solvdlaba/service/impl/CountryServiceImpl.java:39]: @cichan02 javadocs should be only generated for public methods, not for overridden ones

File: src/main/java/by/piskunou/solvdlaba/web/dto/AirportDTO.java (hunk 1 above)
Comment (h1alexbel) [reply src/main/java/by/piskunou/solvdlaba/web/dto/AirportDTO.java:32]: @cichan02 can be suppressed

**Review by h1alexbel — CHANGES_REQUESTED** [review]

File: src/main/java/by/piskunou/solvdlaba/service/impl/CountryServiceImpl.java (hunk 4 above)
Comment (cichan02) [reply src/main/java/by/piskunou/solvdlaba/service/impl/CountryServiceImpl.java:39]: @h1alexbel Removed javadoc from @Overridden methods

File: src/main/java/by/piskunou/solvdlaba/persistence/impl/AirportRepositoryImpl.java [hunk 5]
```diff
@@ -21,52 +21,31 @@ public class AirportRepositoryImpl implements AirportRepository {
     private final MongoAirportRepository repository;
 
     /**
-     * Returns all airports.
-     *
-     * @return a {@link reactor.core.publisher.Flux}
-     * of {@link by.piskunou.solvdlaba.domain.Airport}s
-     * containing all airports.
+     * {@inheritDoc}
```
Comment (h1alexbel) [inline src/main/java/by/piskunou/solvdlaba/persistence/impl/AirportRepositoryImpl.java:24]: @cichan02 You should have no Javadocs on overriden methods

**Review by h1alexbel — CHANGES_REQUESTED** [review]

File: src/main/java/by/piskunou/solvdlaba/persistence/impl/AirportRepositoryImpl.java (hunk 5 above)
Comment (cichan02) [reply src/main/java/by/piskunou/solvdlaba/persistence/impl/AirportRepositoryImpl.java]: @h1alexbel Totally removed docs from overridden methods

**Review by h1alexbel — CHANGES_REQUESTED** [review]

File: checkstyle-suppressions.xml [hunk 6]
```diff
@@ -7,7 +7,7 @@
               files="(Mongo(Airport|City)|ErrorResponseDTO)\.java"
```
[the remark this answers is not in the document]
Comment (h1alexbel) [reply checkstyle-suppressions.xml:7]: @cichan02 ok, but the more classes you will have in the source code the more verbose your `HiddenFiled` suppression will be

[the remark this answers is not in the document]
Comment (cichan02) [reply checkstyle-suppressions.xml:7]: @h1alexbel thanks

**Review by h1alexbel — APPROVED** [review]

**Review by h1alexbel — APPROVED** [review]

**Pull request merged**